students/carlos_novoa/lesson05/mailroom.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+import os
+import datetime
+
+"""
+Lesson4 - Mailroom, Part 3
+"""
+
+now = datetime.datetime.now()
+
+_donors = {
+    'Jim Halpert': [1.00],
+    'Pam Beesley': [1000.00, 2000.00, 3000.00],
+    'Dwight Shrute': [2.00, 3.00],
+    'Michael Scott': [10.00, 20.00, 30.00],
+    'Andy Bernard': [500.00]
+}
+
+
+def quit_menu():
+    return 'break'
+
+
+def thank_donor(donor_name=None, donations=None):
+    if donor_name is None:
+        while True:
+            donor_name = input('Type donor name: ')
+            try:
+                donations = _donors[donor_name]
+            except KeyError:
+                print('Type the exact donor name (see list): ', end='\n\n')
+                list_donors()
+            else:
+                donor_email(donor_name, sum(donations), True)
+                break
+    else:
+        donor_email(donor_name, donations, False)
+
+
+def add_donor():
+    while True:
+        try:
+            donor_name = input('Type donor name: ')
+            if not donor_name:
+                raise ValueError('String required.')
+        except ValueError as e:
+            print(e)
+            continue
+        else:
+            if donor_name in _donors:
+                print('Donor already exists, use append.')
+                return
+            _donors[donor_name] = []
```

Comment:
keep in mind that there is a `defaultdict` available where you can define values to default to a list